doc/source/security/encryption-functions.rst

```diff
@@ -0,0 +1,431 @@
+
+* key_len - the range for the key length is from 1024 to 10,000. The default value is 10,000.
+
+  .. note:: Generating DH keys can take more time than generating RSA or DSA keys. The parameter length limits are defined by OpenSSL. To change the maximum parameter length, use :ref:`dh-threshold`.
+
+.. _create-digest:
+
+create_digest(*digest_type, str*)
+--------------------------------------------
+
+Creates a digest from the given string using the given digest type. The digest string can be used with :ref:`asymmetric_sign<asym-sign>` and :ref:`asymmetric_verify<asym-verify>`.
+
+.. rubric:: Returns
+
+The digest of the given string as a binary string
+
+.. rubric:: Parameters
+
+The parameters are the following:
+
+* digest_type - the supported values are the following (based on OpenSSL version):
+
+  .. list-table::
+      :header-rows: 1
+
+      * - Value Name for OpenSSL 1.0.2
+        - Value Name for OpenSSL 1.1.x addition
+      * - 'MD5'
+        - 'BLAKE2B512'
+      * - 'SHA1'
+        - 'BLAKE2S256'
+      * - 'SHA224'
+        - 'RIPEMD'
+      * - 'SHA256'
+        - 'RMD160'
+      * - 'SHA384'
+        - 'SHAKE128'
+      * - 'SHA512'
+        - 'SHAKE256'
+      * - 'MD4'
+        - 'SM3'
+      * - 'RIPEMD160'
+        - 'WHIRLPOOL'
+
+* str - String used to generate the digest string.
+
+.. _encryption-variables:
+
+.. rubric:: Encryption threshold variables
+
+The key length limits are defined by OpenSSL. Server administrators can set the following MySQL system variables when the server starts to limit the maximum key length. The following variables are automatically registered when `component_encryption_udf` is installed.
+
+Setting the global variable as a runtime value requires either the `SYSTEM_VARIABLES_ADMIN <https://dev.mysql.com/doc/refman/8.0/en/privileges-provided.html#priv_system-variables-admin>`__ or the deprecated `SUPER <https://dev.mysql.com/doc/refman/8.0/en/privileges-provided.html#priv_super>`__ privilege.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Variable Name
+   * - :ref:`dh-threshold`
+
+.. _dh-threshold:
+
+.. rubric:: ``encryption_udf.dh_bits_threshold``
+
+These variable sets the threshold limits for the :ref:`create-dh-param`. The value takes precedence over the OpenSSL maximum length value.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Option
+     - Description
+   * - command-line
+     - Yes
+   * - scope
+     - Global
+   * - data type
+     - unsigned integer
+   * - default
+     - 10,000
+
+The range for this variable is from 1024 to 10,000.
+
+.. _dsa-threshold:
+
+.. rubric:: `encryption_udf.dsa_bits_threshold`
+
+These variables set the threshold limits for :ref:`asymmetric_encrypt <asym-encrypt>`. The value takes precedence over the OpenSSL maximum length value.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Option
+     - Description
+   * - command-line
+     - Yes
+   * - scope
+     - Global
+   * - data type
+     - unsigned integer
+
```

Comment:
* - default
   - 9984
   
   Please add this row for encryption_udf.dsa_bits_threshold

Reply:
Added.